src/main/scala/shine/C/AST/Printer.scala

```diff
@@ -139,7 +139,7 @@ class CPrinter extends Printer {
   private def printParamDecl(p: ParamDecl): Unit = {
     if (p.t.const) print("const ")
     p.t match {
-      case b: BasicType => print(s"${b.name} ${p.name}")
+      case b: BasicType => print(s"${b.print} ${p.name}")
```

Comment:
why this change?

Reply:
The matrixLayout for the Fragment datatype, which are used when using the tensor core primitives, is inferred very late in the codegen. So when the Fragment datatype declaration is insert as basic type into the AST the layout is still unknown (represented as identifier). So the name-attribute of this fragment type with unknown layout is not correct. Using the print-method fix this. Other BasicTypes simply print their names.
It is may a good idea to infer the matrixLayout before translating low level rise to DPIA, but i have no idea how to do this.